ci-operator/step-registry/azure/provision/custom-role/azure-provision-custom-role-commands.sh

```diff
@@ -0,0 +1,385 @@
+"""
+
+
+    # optional permission to gather bootstrap bundle log
+    required_permissions="""
+\"Microsoft.Compute/virtualMachines/retrieveBootDiagnosticsData/action\",
+${required_permissions}
+"""
+
+    # New permissions are instroduced when using CAPZ to provision IPI cluster
+    if [[ "${CLUSTER_TYPE_MIN_PERMISSOIN}" == "IPI" ]] && (( ocp_minor_version >= 17 && ocp_major_version == 4 )); then
+        # routeTables relevant perssions can be removed once OCPBUGS-37663 is fixed.
+        required_permissions="""
+\"Microsoft.Network/routeTables/read\",
+\"Microsoft.Network/routeTables/write\",
+\"Microsoft.Network/routeTables/join/action\",
+\"Microsoft.Network/loadBalancers/inboundNatRules/read\",
+\"Microsoft.Network/loadBalancers/inboundNatRules/write\",
+\"Microsoft.Network/loadBalancers/inboundNatRules/join/action\",
+\"Microsoft.Network/loadBalancers/inboundNatRules/delete\",
+${required_permissions}
+"""
+    fi
+
+    if [[ "${CLUSTER_TYPE_MIN_PERMISSOIN}" == "UPI" ]]; then
+        required_permissions="""
+\"Microsoft.Compute/images/read\",
+\"Microsoft.Compute/images/write\",
+\"Microsoft.Compute/images/delete\",
+\"Microsoft.Compute/virtualMachines/deallocate/action\",
+\"Microsoft.Storage/storageAccounts/blobServices/containers/read\",
+\"Microsoft.Resources/deployments/read\",
+\"Microsoft.Resources/deployments/write\",
+\"Microsoft.Resources/deployments/validate/action\",
+\"Microsoft.Resources/deployments/operationstatuses/read\",
+${required_permissions}
+"""
+    fi
+
+    # optional permissions when installing cluster in existing vnet
+    if [[ -n ${install_config_vnet} ]] && (( ocp_minor_version >= 17 && ocp_major_version == 4 )); then
+        required_permissions="""
+\"Microsoft.Network/virtualNetworks/checkIpAddressAvailability/read\",
+${required_permissions}
+"""
+    fi
+
+    if [[ -n "${install_config_osimage_default}" ]] || [[ -n "${install_config_osimage_master}" ]] || [[ -n "${install_config_osimage_worker}" ]]; then
+        required_permissions="""
+\"Microsoft.MarketplaceOrdering/offertypes/publishers/offers/plans/agreements/read\",
+\"Microsoft.MarketplaceOrdering/offertypes/publishers/offers/plans/agreements/write\",
+\"Microsoft.Compute/images/read\",
+\"Microsoft.Compute/images/write\",
+\"Microsoft.Compute/images/delete\",
+${required_permissions}
+"""    
+    fi
+
+    if [[ -n "${install_config_des_default}" ]] || [[ -n "${install_config_des_master}" ]] || [[ -n "${install_config_des_worker}" ]]; then
+        required_permissions="""
+\"Microsoft.Compute/diskEncryptionSets/read\",
+\"Microsoft.Compute/diskEncryptionSets/write\",
+\"Microsoft.Compute/diskEncryptionSets/delete\",
+\"Microsoft.KeyVault/vaults/read\",
+\"Microsoft.KeyVault/vaults/write\",
+\"Microsoft.KeyVault/vaults/delete\",
+\"Microsoft.KeyVault/vaults/deploy/action\",
+\"Microsoft.KeyVault/vaults/keys/read\",
+\"Microsoft.KeyVault/vaults/keys/write\",
+${required_permissions}
+"""
+    fi
+
+    custom_role_name_json="{}"
```

Comment:
The following `ENABLE_MIN_PERMISSION_FOR_STS` block also need to reference this var.